### PR TITLE
update set-sample-operator step

### PR DESCRIPTION
**Pull request by jianzhangbjz**
@jianlinliu @xiuwang As we discussed yesterday, I updated the related steps, please have a review, thanks! cc: @kuiwang02

### CODE REVIEW

File: ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh [hunk 1]
```diff
@@ -22,38 +22,80 @@ function config_samples_operator()
...
+registry.redhat.io/ubi7/ruby-27:latest=MIRROR_REGISTRY_PLACEHOLDER/ubi7/ruby-27:latest
+registry.redhat.io/rhscl/ruby-25-rhel7:latest=MIRROR_REGISTRY_PLACEHOLDER/rhscl/ruby-25-rhel7:latest
+registry.redhat.io/rhscl/mysql-80-rhel7:latest=MIRROR_REGISTRY_PLACEHOLDER/rhscl/mysql-80-rhel7:latest
+registry.redhat.io/rhel8/mysql-80:latest=MIRROR_REGISTRY_PLACEHOLDER/rhel8/mysql-80:latest
+registry.redhat.io/rhel8/httpd-24:latest=MIRROR_REGISTRY_PLACEHOLDER/rhel8/httpd-24:latest
+" > /tmp/tag_images_list
+
+    sed -i "s/MIRROR_REGISTRY_PLACEHOLDER/${MIRROR_PROXY_REGISTRY}/g" "/tmp/tag_images_list"
+    run_command "cat /tmp/tag_images_list"
+    run_command "cat /tmp/new-dockerconfigjson"
```
Comment (jianlinliu) [inline ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh:94]: Pls not print out the credential in the log.

File: ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh [hunk 2]
```diff
@@ -22,38 +22,80 @@ function config_samples_operator()
...
+    run_command "oc image mirror -f \"/tmp/tag_images_list\"  --insecure=true -a \"/tmp/new-dockerconfigjson\" --skip-missing=true --skip-verification=true --keep-manifest-list=true --filter-by-os='.*'"
+
+}
+
 run_command "oc get config.samples.operator.openshift.io cluster"; ret=$?
```
Comment (jianlinliu) [inline ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh:99]: I think we also need `set_proxy` function before running most of oc command against clusters, Note that, but we do not need proxy when mirror images.

File: ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh [hunk 3]
```diff
@@ -22,38 +22,80 @@ function config_samples_operator()
...
+    optional_auth_password=$(cat "/var/run/vault/mirror-registry/registry_quay.json" | jq -r '.password')
+    quay_auth=`echo -n "${optional_auth_user}:${optional_auth_password}" | base64 -w 0`
+    # # brew.registry.redhat.io
+    # reg_brew_user=$(cat "/var/run/vault/mirror-registry/registry_brew.json" | jq -r '.user')
+    # reg_brew_password=$(cat "/var/run/vault/mirror-registry/registry_brew.json" | jq -r '.password')
+    # brew_registry_auth=`echo -n "${reg_brew_user}:${reg_brew_password}" | base64 -w 0`
+    # vmc.mirror-registry.qe.devcluster.openshift.com:5000
+    registry_cred=`head -n 1 "/var/run/vault/mirror-registry/registry_creds" | base64 -w 0`
+    # registry.redhat.io
+    reg_user=$(cat "/var/run/vault/mirror-registry/registry.json" | jq -r '.user')
```
Comment (jianlinliu) [inline ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh:44]: How about renaming `registry.json` to `registry_redhat.json` to be more easy-reading.

File: ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh [hunk 4]
```diff
@@ -22,38 +22,80 @@ function config_samples_operator()
...
+    reg_password=$(cat "/var/run/vault/mirror-registry/registry.json" | jq -r '.password')
+    registry_auth=`echo -n "${reg_user}:${reg_password}" | base64 -w 0`
 
-    sed -i "s/MIRROR_REGISTRY_PLACEHOLDER/${REGISTRY_HOST}:5000/g" "/tmp/tag_images_list"
-    run_command "cat /tmp/tag_images_list"
-    run_command "oc image mirror -f \"/tmp/tag_images_list\"  --insecure=true -a \"/tmp/.dockerconfigjson\" --skip-missing=true --skip-verification=true --keep-manifest-list=true --filter-by-os='.*'"; ret=$?
+    jq --argjson a "{\"registry.redhat.io\": {\"auth\": \"${registry_auth}\"}, \"quay.io\": {\"auth\": \"$quay_auth\"}, \"${MIRROR_PROXY_REGISTRY}\": {\"auth\": \"$registry_cred\"}}" '.auths |= . + $a' "/tmp/.dockerconfigjson" > /tmp/new-dockerconfigjson
```
Comment (jianlinliu) [inline ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh:48]: I just go though the installer mirror steps, seem like we do not have to get pull secret for quay.io, redhat.io from vault, all the pull secrets are already  saved in `${CLUSTER_PROFILE_DIR}/pull-secret`, hence we can directly use `${CLUSTER_PROFILE_DIR}/pull-secret` for mirror command, once that, we can drop `config_mirror_auth`.

File: ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh [hunk 5]
```diff
@@ -62,8 +104,9 @@ else
     return 0
 fi
 
+# vmc.mirror-registry.qe.devcluster.openshift.com:5000
 MIRROR_PROXY_REGISTRY=`head -n 1 "${SHARED_DIR}/mirror_registry_url"`
-# get registry host name, no port
-REGISTRY_HOST=`echo ${MIRROR_PROXY_REGISTRY} | cut -d \: -f 1`
+set_CA_for_nodes
+config_mirror_auth
 mirror_tag_images
```
Comment (jianlinliu) [inline ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh:111]: @xiuwang do you know if sample co get healthy only mirror these images, but not creating icsp for 6001 and 6002 mirror registry?

Comment (xiuwang) [reply ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh:111]: Correct, samples co will not depend on icsp, the openshift imagesteams point to mirror registry images with tag not used the icsp.

**Comment by kuiwang02** [comment]
https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/30717/rehearse-30717-periodic-ci-openshift-openshift-tests-private-release-4.11-nightly-4.11-upgrade-from-stable-4.10-azure-ipi-disconnected-fullyprivate/1550312327620857856 and https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/30717/rehearse-30717-periodic-ci-openshift-openshift-tests-private-release-4.11-nightly-4.11-upgrade-from-stable-4.10-gcp-ipi-disconnected-private/1550312325364322304 are still failing at this step. please check it. thanks

File: ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh (hunk 1 above)
Comment (jianzhangbjz) [reply ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh:94]: will remove it once test passed.

File: ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh (hunk 2 above)
Comment (jianzhangbjz) [reply ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh:99]: Good catch! Will have a try.

File: ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh (hunk 4 above)
Comment (jianzhangbjz) [reply ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh:48]: Yeah, but remaining this `config_mirror_auth` will make this step doesn't depend on any other steps.

Comment (jianzhangbjz) [reply ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh:48]: Anyway, updated it.

File: ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh [hunk 6]
```diff
@@ -62,8 +115,13 @@ else
     return 0
 fi
 
+set_proxy
```
Comment (jianlinliu) [inline ci-operator/step-registry/set-sample-operator/disconnected/set-sample-operator-disconnected-commands.sh:118]: I think this line need to be moved to line 110

**Comment by jianlinliu** [comment]
> Get the below error, set `config_mirror_auth` to have a try.
> 
> ```yaml
>   stats: shared=25 unique=65 size=3.033GiB ratio=0.63
> 
> error: unable to retrieve source image quay.io/openshift-qe-optional-operators/ocp4-index by tag latest: unauthorized: access to the requested resource is not authorized
> error: an error occurred during planning
> ```

Other images on redhat.io can be retrieved successfully, sounds like the quay.io pullsecret in CLUSTER_PROFILE does not have access permission to openshift-qe-optional-operators/ocp4-index namespace. 

BTW, is openshift-qe-optional-operators/ocp4-index  is a required image to upload?

**Comment by jianzhangbjz** [comment]
> BTW, is openshift-qe-optional-operators/ocp4-index is a required image to upload?

For OLM operators, no, but I'm not sure if any other teams need it from the proxy-registry:5000.  Anyway, I add the auth of quay.io/openshift-qe-optional-operators to have a try.

**Comment by jianlinliu** [comment]
> > BTW, is openshift-qe-optional-operators/ocp4-index is a required image to upload?
> 
> For OLM operators, no, but I'm not sure if any other teams need it from the proxy-registry:5000. Anyway, I add the auth of quay.io/openshift-qe-optional-operators to have a try.

From enable-qe-catalogsource-disconnected, I saw the ICSP for quay.io/openshift-qe-optional-operators is already created, I guess we do not need to upload it any more in mirror_tag_images

**Comment by jianzhangbjz** [comment]
@xiuwang Do you happen to know will the `IS` use this quay.io/openshift-qe-optional-operators/ocp4-index:latest? Thanks!

**Comment by jianlinliu** [comment]
> @xiuwang Do you happen to know will the `IS` use this quay.io/openshift-qe-optional-operators/ocp4-index:latest? Thanks!

I guess IS never use it, maybe @anpingli know more about it.

**Comment by jianzhangbjz** [comment]
> I guess IS never use it, maybe @anpingli know more about it.

Ok, I will remove it. Will add it later if any sub-teams need it.
Based on the below error, I will add the auth of the `proxy registry`.
```yaml
error: unable to upload blob sha256:99ed79bea001b5d70fe56b765504b33223c5353df984e880d51daad80d359564 to ci-op-tytpfkgq-75190.mirror-registry.XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX:5000/ubi7/ruby-27: Post "https://ci-op-tytpfkgq-75190.mirror-registry.XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX:5000/v2/ubi7/ruby-27/blobs/uploads/": no basic auth credentials
info: Mirroring completed in 6.96s (0B/s)
```

**Comment by jianzhangbjz** [comment]
Passed: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/30717/rehearse-30717-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.10-installer-rehearse-aws/1550430378878046208/artifacts/installer-rehearse-aws/set-sample-operator-disconnected/build-log.txt

**Comment by jianzhangbjz** [comment]
@xiuwang @kuiwang02 @bandrade Could you help add `/lgtm` label? Thanks!

**Pull request merged**